rbac/management/group/relation_api_dual_write_group_handler.py

```diff
@@ -0,0 +1,110 @@
+#
+# Copyright 2024 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Class to handle Dual Write API related operations."""
+import logging
+from typing import Optional
+
+from django.conf import settings
+from management.role.relation_api_dual_write_handler import (
+    DualWriteException,
+    OutboxReplicator,
+    RelationReplicator,
+    ReplicationEvent,
+    ReplicationEventType,
+)
+from migration_tool.utils import create_relationship
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class RelationApiDualWriteGroupHandler:
+    """Class to handle Dual Write API related operations."""
+
+    def __init__(
+        self,
+        group,
+        event_type: ReplicationEventType,
+        principal_usernames,
+        service_accounts,
```

Comment:
Since these are both just `List[Principal]` what do you think about combining the parameters? It makes it a little easier to use (e.g. in tests) because I don't have to explicitly sort users vs services ahead of time, and its not actualy needed as far as I can tell since we combine them into one below (`for principal in self.principal_usernames + self.service_accounts`).

Reply:
I did not combine this initially because I wasn't sure if we would need separate IDs for the service account and principals but as you mentioned it seems unnecessary for now. Anyway If it becomes necessary later, we can handle it in the dual write handler.

I updated also your tests and I removed service account creation as well - but if you intended to keep it for later use  we can revert/remove last commit.

Reply:
K cool! Let's keep the service account creation support in the test fixture – that is still a use case we want to test explicitly I think, regardless of the implementation.

As for IDs, a fair point. I am thinking both will end up using "user_id" field being added in #1179 though, so we might be okay.